src/main/java/com/powsybl/openloadflow/ac/outerloop/TransformerVoltageControlOuterLoop.java

```diff
@@ -99,62 +70,87 @@ public String getName() {
-        // disabled previously are enabled.
-        if (context.getIteration() == 1) {
-            secondOuterLoop(context, status, contextData);
-        }
+        return switch (contextData.step) {
+            case INITIAL -> initStep(context.getNetwork(), contextData);
+            case CONTROL -> controlStep(context.getNetwork(), contextData);
+            case COMPLETE -> new OuterLoopResult(this, OuterLoopStatus.STABLE);
+        };
 
-        return new OuterLoopResult(this, status.getValue());
     }
 
-    private static void firstOuterLoop(AcOuterLoopContext context, ContextData contextData, MutableObject<OuterLoopStatus> status) {
-        double maxControlledNominalVoltage = contextData.getMaxControlledNominalVoltage();
-        for (LfBus bus : context.getNetwork().getControlledBuses(VoltageControl.Type.GENERATOR)) {
-            if (bus.getNominalV() <= maxControlledNominalVoltage) {
-                var voltageControl = bus.getGeneratorVoltageControl().orElseThrow();
-                voltageControl.getMergedControllerElements().forEach(controllerBus -> {
-                    if (controllerBus.isGeneratorVoltageControlEnabled()) {
-                        controllerBus.setGenerationTargetQ(controllerBus.getQ().eval());
-                        controllerBus.setGeneratorVoltageControlEnabled(false);
-                        contextData.getBusesWithVoltageControlDisabled().add(controllerBus);
-                    }
-                });
-                status.setValue(OuterLoopStatus.UNSTABLE);
-            }
-        }
-        for (LfBranch branch : context.getNetwork().<LfBranch>getControllerElements(VoltageControl.Type.TRANSFORMER)) {
-            branch.getVoltageControl().ifPresent(voltageControl -> {
+    /**
+     * At first outer loop iteration, the voltage control of generators that controlled under the max controlled nominal
+     * voltage are disabled (automatic detection or parameter).
+     * The transformer voltage controls are enabled and their continuous ratio is computed.
+     * @return a stable status if all taps are already tuned for tension control, unstable otherwise
+     */
+    private OuterLoopResult initStep(LfNetwork network, ContextData contextData) {
+        boolean needRun = false;
+        for (LfBranch branch : network.<LfBranch>getControllerElements(VoltageControl.Type.TRANSFORMER)) {
+            if (branch.getVoltageControl().isPresent()) {
+                TransformerVoltageControl voltageControl = branch.getVoltageControl().orElseThrow();
                 double targetV = voltageControl.getTargetValue();
                 double v = voltageControl.getControlledBus().getV();
                 double diffV = targetV - v;
                 double halfTargetDeadband = getHalfTargetDeadband(voltageControl);
                 if (Math.abs(diffV) > halfTargetDeadband && branch.isConnectedAtBothSides()) {
                     branch.setVoltageControlEnabled(true);
```

Comment:
What should happen to the ones inside their deadband and already with voltage control enabled?

Reply:
They cannot be, see above:
````
        for (LfBranch controllerBranch : context.getNetwork().<LfBranch>getControllerElements(VoltageControl.Type.TRANSFORMER)) {
            controllerBranch.setVoltageControlEnabled(false);
        }
````

Reply:
Yes, I see that in the initialize()